### PR TITLE
a minimalist implementation of CoRIM / CoMID

### LINKED ISSUE #1: Provide a minimal implementation that allows provisioning PSA endorsements
A minimalist implementation of CoRIM / CoMID that can be used to ship PSA endorsements.

In particular, we need CBOR encoding and decoding, JSON encoding to ease creation of CoRIM/CoMID by humans.

Obviously, also all the PSA extension points and enough API surface to make it easy to extract PSA endorsements from CoRIM/CoMID.

### PR DESCRIPTION
**Pull request by thomas-fossati**
This PR includes:
* CoMID: implementation of full CBOR encoder and decoder, the JSON decoder (for easy templating by humans), the PSA extractors;
* Rough but working implementation of unsigned CoRIM.

Next steps: signed CoRIM using the COSE library, documentation, and more test coverage. 

Fixes #1

### CODE REVIEW

**Comment by thomas-fossati** [comment]
https://github.com/ietf-rats/ietf-corim-cddl/issues/121

File: comid/classid.go [hunk 1]
```diff
@@ -0,0 +1,196 @@
+package comid
+
+// TODO(tho) OID
```
Comment (thomas-fossati) [inline comid/classid.go:3]: this is done now
```suggestion
```

File: comid/classid.go [hunk 2]
```diff
@@ -0,0 +1,196 @@
...
+
+import (
+	"encoding/base64"
+	"encoding/json"
+	"fmt"
+)
+
+// ClassID represents a $class-id-type-choice, which can be one of TaggedUUID,
+// TaggedOID (TODO), or TaggedImplID (PSA-specific extension)
```
Comment (thomas-fossati) [inline comid/classid.go:12]:
```suggestion
// TaggedOID, or TaggedImplID (PSA-specific extension)
```

File: comid/class.go [hunk 3]
```diff
@@ -0,0 +1,153 @@
...
+
+import (
+	"encoding/json"
+	"fmt"
+)
+
+// TODO(tho) OID
+
+// Class represents the class of the (target / attesting) environment.  The only
+// required field is the class unique identifier (see ClassID).  Optionally,
```
Comment (yogeshbdeshpande) [inline comid/class.go:11]: is line 15 strictly required, I mean the class identifier.

I thought this is also optional, as the device class could also be mapped using V,M.L and I.

File: comid/class.go [hunk 4]
```diff
@@ -0,0 +1,153 @@
...
+		ClassID: &ClassID{},
+	}
+
+	if c.ClassID.SetUUID(uuid) == nil {
+		return nil
+	}
+	return &c
+}
+
+// NewClassUUID instantiates a new Class object that identifies the specified PSA
```
Comment (yogeshbdeshpande) [inline comid/class.go:35]: you mean NewClassImplID here?

File: comid/psareferencevalue.go [hunk 5]
```diff
@@ -0,0 +1,64 @@
+package comid
+
+import (
+	"fmt"
+)
+
```
Comment (yogeshbdeshpande) [inline comid/psareferencevalue.go:6]: Perhaps this comment/question is applicable on a much more top level:

I was thinking, will it be more logically convenient to implement 
(a) Base CoMID and Base CoRIM under CoMID and CoRIM golang Packages

Any PSA specific Profile be extended inside a PSA package, outside of base packages.
Would like to discuss this aspect in our code walkthrough today to hear other opinions on the same!

Comment (yogeshbdeshpande) [reply comid/psareferencevalue.go:9]: https://github.com/veraison/veraison/issues/65 tracks the issue!

**Review by yogeshbdeshpande — APPROVED** [review]

**Review by setrofim — APPROVED** [review]
LGTM

**Pull request merged**